@here/harp-datasource-protocol/lib/InterpolatedProperty.ts

```diff
@@ -85,14 +111,45 @@ export function isInterpolatedProperty(p: any): p is InterpolatedProperty {
  * interpolation of `length` values).
  *
  */
-export function getPropertyValue<T>(
+export function getPropertyValue(
     property: Value | Expr | InterpolatedProperty | undefined,
     level: number,
+    pixelToMeters?: number
+): any;
+
+export function getPropertyValue(
+    property: Value | Expr | InterpolatedProperty | undefined,
+    envOrLevel: number | Env,
     pixelToMeters: number = 1.0
 ): any {
-    if (isInterpolatedPropertyDefinition<T>(property)) {
+    if (isInterpolatedPropertyDefinition(property)) {
         throw new Error("Cannot interpolate a InterpolatedPropertyDefinition.");
-    } else if (!isInterpolatedProperty(property)) {
+    }
+
+    if (Expr.isExpr(property)) {
+        let env: Env;
+
+        if (typeof envOrLevel === "number") {
+            dynamicPropertiesTempEnv.entries.$zoom = envOrLevel;
+            dynamicPropertiesTempEnv.entries.$pixelToMeters = pixelToMeters;
+            env = dynamicPropertiesTempEnv;
+        } else {
+            env = envOrLevel;
+        }
+
+        return property.evaluate(env);
+    }
+
+    let level: number;
+
+    if (typeof envOrLevel === "number") {
+        level = envOrLevel;
+    } else {
+        level = envOrLevel.lookup("$zoom") as any;
+        pixelToMeters = envOrLevel.lookup("$pixelToMeters") as any;
+    }
```

Comment:
`$zoom` and and `$pixelToRatio` are guaranteed to be in the `Env` and have "number" value, so there's no need for additional checks here.

Reply:
> guaranteed
Maybe use typescript or at least assert for this guarantee ?

How about new `IntepolationEnv` (accepted by `getPropertyValue)` which has first class `zoom` and `pixelToMeters` members.

Reply:
what do you mean?

Reply:
nvm, please leav as it is as this is only temporary.